src/algorithms/optimization/fixed_point_differentiation.jl

```diff
@@ -1,99 +1,167 @@
 abstract type GradMode{F} end
 
+const GRADIENT_MODE_SYMBOLS = IdDict{Symbol,Type{<:GradMode}}()
+const LINSOLVER_SOLVER_SYMBOLS = IdDict{Symbol,Type{<:KrylovKit.LinearSolver}}(
+    :gmres => GMRES, :bicgstab => BiCGStab
+)
+const EIGSOLVER_SOLVER_SYMBOLS = IdDict{Symbol,Type{<:KrylovKit.KrylovAlgorithm}}(
+    :arnoldi => Arnoldi
+)
+
+function GradMode(;
+    alg=Defaults.gradient_alg,
+    tol=Defaults.gradient_tol,
+    maxiter=Defaults.gradient_maxiter,
+    verbosity=Defaults.gradient_verbosity,
+    iterscheme=Defaults.gradient_iterscheme,
+    solver_alg=(;),
+)
+    # replace symbol with GradMode alg type
+    haskey(GRADIENT_MODE_SYMBOLS, alg) ||
+        throw(ArgumentError("unknown GradMode algorithm: $alg"))
+    alg_type = GRADIENT_MODE_SYMBOLS[alg]
+
+    # parse GradMode algorithm
+    gradient_algorithm = if alg_type <: Union{GeomSum,ManualIter}
+        alg_type{iterscheme}(tol, maxiter, verbosity)
+    elseif alg_type <: Union{<:LinSolver,<:EigSolver}
+        solver = if solver_alg isa NamedTuple # determine linear/eigen solver algorithm
+            solver_kwargs = (; tol, maxiter, verbosity, solver_alg...)
+
+            solver_type = if alg_type <: LinSolver # replace symbol with solver alg type
+                solver_kwargs = (; alg=Defaults.gradient_linsolver, solver_kwargs...)
+                haskey(LINSOLVER_SOLVER_SYMBOLS, solver_kwargs.alg) || throw(
+                    ArgumentError("unknown LinSolver solver: $(solver_kwargs.alg)"),
+                )
+                LINSOLVER_SOLVER_SYMBOLS[solver_kwargs.alg]
+            elseif alg_type <: EigSolver
+                solver_kwargs = (;
+                    alg=Defaults.gradient_eigsolver,
+                    eager=Defaults.gradient_eigsolver_eager,
+                    solver_kwargs...,
+                )
+                haskey(EIGSOLVER_SOLVER_SYMBOLS, solver_kwargs.alg) || throw(
+                    ArgumentError("unknown EigSolver solver: $(solver_kwargs.alg)"),
+                )
+                EIGSOLVER_SOLVER_SYMBOLS[solver_kwargs.alg]
+            end
+
+            solver_kwargs = Base.structdiff(solver_kwargs, (; alg=nothing)) # remove `alg` keyword argument
+            solver_type(; solver_kwargs...)
+        else
+            solver_alg
+        end
+
+        alg_type{iterscheme}(solver)
+    else
+        throw(ArgumentError("unknown gradient algorithm: $alg"))
+    end
+
+    return gradient_algorithm
+end
+
 iterscheme(::GradMode{F}) where {F} = F
 
 """
-    struct GeomSum(; maxiter=Defaults.fpgrad_maxiter, tol=Defaults.fpgrad_tol,
-                   verbosity=0, iterscheme=Defaults.iterscheme) <: GradMode{iterscheme}
+    struct GeomSum <: GradMode{iterscheme}
```

Comment:
Same comment, I would add a line
```julia
GeomSum(; kwargs...)
```
to the signature so people know there's a keyword-only call.